notebooks/02-ingestion/05-partitioning-data.ipynb

```diff
@@ -0,0 +1,717 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "0cb3b009-ebde-4d56-9d59-a028d66d8309",
+   "metadata": {},
+   "source": [
+    "# Optimizing Performance using PARTITIONED BY and CLUSTERED BY\n",
+    "<!--\n",
+    "  ~ Licensed to the Apache Software Foundation (ASF) under one\n",
+    "  ~ or more contributor license agreements.  See the NOTICE file\n",
+    "  ~ distributed with this work for additional information\n",
+    "  ~ regarding copyright ownership.  The ASF licenses this file\n",
+    "  ~ to you under the Apache License, Version 2.0 (the\n",
+    "  ~ \"License\"); you may not use this file except in compliance\n",
+    "  ~ with the License.  You may obtain a copy of the License at\n",
+    "  ~\n",
+    "  ~   http://www.apache.org/licenses/LICENSE-2.0\n",
+    "  ~\n",
+    "  ~ Unless required by applicable law or agreed to in writing,\n",
+    "  ~ software distributed under the License is distributed on an\n",
+    "  ~ \"AS IS\" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY\n",
+    "  ~ KIND, either express or implied.  See the License for the\n",
+    "  ~ specific language governing permissions and limitations\n",
+    "  ~ under the License.\n",
+    "  -->\n",
+    "\n",
+    "This notebook is a review of Apache Druid's data sharding strategy and how it can be used to improve system health and query performance.\n",
```

Comment:
I agree, I used "sharding" because I'm kinda wary of "partitions" meaning "segment files", "segments" meaning "partitions" or "time chunks", ... and so on. The taxonomy and associated nomenclature I understand is:
"segment granularity" - defines the size of the time intervals that group a set of "segment files" 
"segment file" - is a single file or partition of the data, which corresponds to the intersection of segment granularity and partitioning strategy
"secondary partitioning" - a way of splitting the data into segment files that correspond to the same "time chunk", can be triggered by "partitionSpec" in native batch or "CLUSTERED BY". In none are or the ingestion is appending with appendToExisting=true in native or SQL based INSERT specified, then dynamic partitioning is used which just means, fill up a segment file with N rows and then create the next one. 
So with "sharding" I am referring to all of the above.  Should we say "data splitting mechanisms", "data organization strategy"?

Reply:
I changed it to "data organization strategy" but there are references to "sharding" here: https://druid.apache.org/docs/latest/design/segments#sharding